### ISSUE TITLE
E2E: Automate updating of all plugins in nightly and release smoke test sites at a certain time

### ISSUE THREAD

**Issue by rodelgc** [woothemes/woocommerce#36710]
### Describe the solution you'd like

As discussed in pcr5Cj-46L-p2#comment-3695, one way to ensure E2E tests from failing in the nightly and release smoke test sites is to disable automatic updating of plugins in them, and then update them (via cron, GitHub actions, etc.) at a certain time that we think will not coincide with the "Smoke test daily" and "Smoke test release" workflows.

**Comment by juliaamosova** [comment]
Noting that we looked at this issue recently as a team and it is still valid. The conversation linked in the issue pcr5Cj-46L-p2#comment-3695 is still relevant and provides details about the issues we are seeing:

- Remaining flakiness in E2E tests

  - Since the nightly smoke test site runs the tests slower than localhost, some tests that do not run in parallel in localhost run in parallel here, and that’s the time when we discover that they interfere with each other. Examples include tests that enable and disable COD, certain shipping settings, etc., that don’t normally interfere with each other in localhost.
  - To fix this, I’ll create individual tasks to fix each conflict in the E2E tests. It might take a bit of time to eliminate all the flakiness since new conflicts might be discovered as new set of tests get run in parallel in the nightly smoke test site.
- Admin email verification screen

  - Occasionally, Smoke test daily fails because the [Site Admin Email Verification Screen](https://href.li/?https://make.wordpress.org/core/2019/10/17/wordpress-5-3-admin-email-verification-screen/) gets displayed upon logging in.
  - Possible workaround [#1](https://solarisp2.wordpress.com/tag/1/): Disable this screen through the admin_email_check_interval filter as suggested [here](https://href.li/?https://make.wordpress.org/core/2019/10/17/wordpress-5-3-admin-email-verification-screen/).
  - Possible workaround [#2](https://solarisp2.wordpress.com/tag/2/): Set the admin_email_check_interval filter to a far less frequent interval, like once a year (WP documentation says the default is every 6 months, but I’m not sure if that’s what our nightly smoke test site follows.)
  - Possible workaround [#3](https://solarisp2.wordpress.com/tag/3/): Add an extra step in [Check daily smoke test site status](https://href.li/?https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily-site-check.yml) to login to the WP Admin Dashboard and check if this screen appears (maybe through a k6 script so we don’t need to go through the UI?). If it does, send a Slack message on the next step in that workflow (Send message to Slack API) so that the tools porter (or anyone in our team) could confirm and dismiss this screen.

- Automatic plugin updates in the nightly smoke test site

  - Smoke test daily fails sometimes because it runs while some of the other plugins were updating.
  - Possible workaround [#1](https://solarisp2.wordpress.com/tag/1/): Disable automatic updating of plugins in both the nightly and release smoke test sites, which I’ve already done for now. Disadvantage: Additional work on our side to manually update the plugins in them.
  - Occasionally getting 500 errors that lasts throughout the 4 E2E retries. Not much we can do about this except for the tools porter to manually do a rerun of Smoke test daily

**Comment by adimoldovan** [comment]
I'm closing this issue because it's no longer relevant.
The nightly checks are only using wp-env currently where this is not an issue.
New runs against existing environments are being implemented from scratch with a new approach on keeping the setup clean and maintainable.
